File: VioletMarketplace/src/main/java/com/skilldistillery/marketplace/services/BidService.java
package com.skilldistillery.marketplace.services;

import com.skilldistillery.marketplace.entities.Bid;

import java.util.List;

public interface BidService {

    List<Bid> userBids(int userId);

    boolean destroyBid(int bidId);

    Bid create(Bid bid);
}

File: VioletMarketplace/src/main/java/com/skilldistillery/marketplace/services/BidService.kt
package com.skilldistillery.marketplace.services

import com.skilldistillery.marketplace.entities.Auction
import com.skilldistillery.marketplace.entities.Bid

interface BidService {
    fun userBids(userId: Int): Set<Bid>
    fun destroyBid(bidId: Int): Boolean
    fun placeBid(bid: Bid, auction: Auction): Bid
}
